Documents minor differences between ROS1 and ROS2 bloom releases.

nuclearsandwich opened the pull request:
Continuing from https://github.com/ros2/ros2_documentation/pull/2030

This is an update of that pull request which includes links to the ros2-gbp-github-org management repository and a more detailed explanation of why release repositories for ROS 2 are maintained in a single organization.

nuclearsandwich commented:
FYI @norro

clalancette commented on source/How-To-Guides/Releasing-a-ROS-2-package-with-bloom.rst:
```diff
@@ -44,10 +44,21 @@ If you're using a version of bloom older than 0.6.8 you'll need to use the v3 in
+----------------------------------
+
+If you've bloomed packages before in ROS 1, the ROS 2 process should be familiar to you.
+The major difference is that release repositories for ROS 2 packages live in a dedicated github organization:
```
```suggestion
The major difference is that release repositories for ROS 2 packages live in a dedicated GitHub organization:
```

clalancette commented on source/How-To-Guides/Releasing-a-ROS-2-package-with-bloom.rst:
```diff
@@ -44,10 +44,21 @@ If you're using a version of bloom older than 0.6.8 you'll need to use the v3 in
+If you've bloomed packages before in ROS 1, the ROS 2 process should be familiar to you.
+The major difference is that release repositories for ROS 2 packages live in a dedicated github organization:
+`ROS 2 release repositories <https://github.com/ros2-gbp/>`.
+In order to take advantage of new automation supporting the `Rolling distribution <../Releases/Release-Rolling-Ridley>` Open Robotics recommends that release repositories for ROS 2 are hosted in the dedicated `ros2-gbp GitHub organization <https://github.com/ros2-gbp/>`.
```
This sentence is a bit redundant with the previous one.  Maybe something more like:

```suggestion
The dedicated organization allows new automation supporting the :doc:`Rolling distribution <../Releases/Release-Rolling-Ridley>`.
```

clalancette reviewed (changes requested):
I've left some minor fixes, but otherwise looks good to me.

nuclearsandwich commented:
Took each suggestion and applied the backport-all label to let mergify do its thing.

clalancette reviewed (approved):
I fixed the CI issues, and also updated the link at the end to use the `{DISTRO}`/`{DISTRO_TITLE}` macros to generate a URL appropriate to this release.  With all of that, and CI going green again, looks good to me.

nuclearsandwich commented:
> I fixed the CI issues, and also updated the link at the end to use the `{DISTRO}`/`{DISTRO_TITLE}` macros to generate a URL appropriate to this release. With all of that, and CI going green again, looks good to me.

Thanks Chris!
